spark-clickhouse-connector/src/main/scala/io/clickhouse/spark/connector/partitioner/ClickhousePartition.scala

```diff
@@ -18,9 +18,9 @@ case class DateRange(dated: DateTime,
 
   def sql(): String = {
     if (rType == RangeType.HOUR)
-      s"$pk = '${dated.hourOfDay.roundFloorCopy.toString("yyyy-MM-dd HH:mm:ss")}'"
+      s"toStartOfHour($pk) = '${dated.hourOfDay.roundFloorCopy.toString("yyyy-MM-dd HH:mm:ss")}'"
     else
-      s"$pk = '${dated.toString("yyyy-MM-dd")}'"
+      s"toYYYYMMDD($pk) = '${dated.toString("yyyyMMdd")}'"
```

Comment:
There's an error. Need to remove the single quotes because the result of toYYYYMMDD func is Int.